From 88ccf1703c32cf5a7a5d1535172477607fcdddf2 Mon Sep 17 00:00:00 2001
From: jbelloua <jean-marie.bellouard@intel.com>
Date: Fri, 13 Mar 2015 17:02:03 +0100
Subject: [PATCH] [BT] add support for snowfieldspeak

Add support for Snowfieldpeak device in kernel.
It has no host wake gpio.

Change-Id: I9ce4282bab33ac172b153f4336423fe96f9e710b
Signed-off-by: jbelloua <jean-marie.bellouard@intel.com>
---
 drivers/acpi/acpi_platform.c      |  1 +
 drivers/misc/intel_ld/lnp_ldisc.c | 13 ++++++++++++-
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/drivers/acpi/acpi_platform.c b/drivers/acpi/acpi_platform.c
index 7abcd31..9064648 100644
--- a/drivers/acpi/acpi_platform.c
+++ b/drivers/acpi/acpi_platform.c
@@ -34,6 +34,7 @@ static const struct acpi_device_id acpi_platform_device_ids[] = {
 	{ "LNV4752" },
 	{ "BCM2E1A" },
 	{ "INT33E1" },
+	{ "INTL1216" },
 	{ "BCM2E39" },
 	{ "BCM2E3D" },
 	{ "BCM2E3A" },
diff --git a/drivers/misc/intel_ld/lnp_ldisc.c b/drivers/misc/intel_ld/lnp_ldisc.c
index ebfea73..38d3f41 100644
--- a/drivers/misc/intel_ld/lnp_ldisc.c
+++ b/drivers/misc/intel_ld/lnp_ldisc.c
@@ -42,15 +42,22 @@ struct intel_bt_gpio_desc {
 	int			reg_on_idx;
 	int			host_wake_idx;
 };
+
 static struct intel_bt_gpio_desc acpi_default_bluetooth = {
 	.reg_on_idx = 1,
 	.host_wake_idx = 0,
 };
 
+/* snowfieldpeak: not host wake*/
+static struct intel_bt_gpio_desc acpi_sfp_bluetooth = {
+	.reg_on_idx = 0,
+	.host_wake_idx = -1,
+};
 
 static struct acpi_device_id intel_id_table[] = {
 	/* ACPI IDs here */
 	{ "INT33E1", (kernel_ulong_t)&acpi_default_bluetooth },
+	{ "INTL1216", (kernel_ulong_t)&acpi_sfp_bluetooth },
 	{ },
 };
 
@@ -2038,8 +2045,12 @@ static void lbf_ldisc_receive(struct tty_struct *tty, const u8 *cp, char *fp,
 				if (proto != INVALID) {
 					lbf_uart->rx_skb = alloc_skb(1700,
 						GFP_ATOMIC);
-					if (!lbf_uart->rx_skb)
+					if (!lbf_uart->rx_skb) {
+						spin_unlock_irqrestore(
+							&lbf_uart->rx_lock,
+							flags);
 						return;
+					}
 					lbf_uart->rx_chnl = proto;
 					lbf_uart->rx_state = LBF_W4_PKT_HDR;
 					lbf_uart->rx_count =
-- 
1.9.1

